src/components/renderer/field/form-field-renderer.component.tsx

```diff
@@ -76,6 +76,16 @@ export const FormFieldRenderer = ({ field, valueAdapter, repeatOptions }: FormFi
         console.error(error);
       }
     }
+
+    if (field.type === 'programState' && field.questionOptions.enablePreviousValue) {
+      try {
+        context.processor.getHistoricalValue(field, context).then((value) => {
+          setHistoricalValue({value: value.value, display: value.display});
+        });
+      } catch (error) {
+        console.error(error);
+      }
+    }
```

Comment:
Seems like historical values are handled in the enclosure or code block above, this seems like a duplication.

Reply:
Are you able to see the value when you test on your end?

Reply:
Note that this is currently restricted to working in "enter" mode. That could the reason as to why it's not working for you. I however don't object lifting this constraint.

Reply:
okay